components/search_results_header/index.jsx

```diff
@@ -8,16 +8,24 @@ import {
     closeRightHandSide,
     toggleRhsExpanded,
 } from 'actions/views/rhs';
+import {getIsRhsExpanded} from 'selectors/rhs';
 
 import SearchResultsHeader from './search_results_header.jsx';
 
+function mapStateToProps(state) {
+    return {
+        isExpanded: getIsRhsExpanded(state),
+    };
+}
+
 function mapDispatchToProps(dispatch) {
     return {
         actions: bindActionCreators({
             closeRightHandSide,
             toggleRhsExpanded,
+            dispatch: (action) => (action()), // Used to dispatch arbitrary action
```

Comment:
Could we pass in the icons action already bound to dispatch from the plugin like we do with the channel header icons? Not a fan of passing a generic dispatch to the component and we can keep the convention of how plugins have been passing actions up to now.

Reply:
Since we can have an arbitrary number of `icons`, i'll have to have a mapping of icon->icon's action, which is generated somewhere, which is not ideal either. Also this is not passing dispatch down to the component, this is passing a function that accepts an action that _will_ be dispatched in the redux layer. So not really exposing dispatch to the component. I just happen to name it that, I can name it something different like `dispatchAction`?